### PR TITLE
Upgrade to Spring Boot 3.2

### PR DESCRIPTION
**Pull request by arolfes**
---
Release Notes:

```
Breaking:
* Upgrade to Spring Boot 3.2.x
```
 
### For the submitter:
- [ ] I updated the [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview) and will supply links to the specific files
- [ ] I did not update the [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview)
- [ ] I included a link to the [SonarCloud branch analysis](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1019969636/SonarCloud+Integration)
- [ ] After integration of the PR, I added a description of changes on the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana)
- [ ] I did not update the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana)
- [ ] I put the ticket in review
- [ ] After integration of the pull request, I verified our [bluemix test environment](http://taskana.mybluemix.net/taskana) is not broken

### Verified by the reviewer:
- [ ] Commit message format → (Closes) #&lt;Issue Number&gt;: Your commit message.
- [ ] Submitter's update to [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview) is sufficient
- [ ] SonarCloud analysis meets our standards
- [ ] Update of the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana) reflects changes
- [ ] PR fulfills the ticket
- [ ] Edge cases and unwanted side effects are tested
- [ ] Readability

### CODE REVIEW

**Review by mustaphazorgati — COMMENTED** [review]
LGTM!

Did you forget to remove the 'xxx-rest-api.adoc' files?

**Comment by arolfes** [comment]
Yes
Because I didn't know it

**Comment by arolfes** [comment]
we decided to delete the files later

File: rest/taskana-rest-spring/src/main/java/pro/taskana/common/rest/TaskanaRestExceptionHandler.java [hunk 1]
```diff
@@ -117,13 +117,22 @@ public class TaskanaRestExceptionHandler extends ResponseEntityExceptionHandler
           Map.entry(UnsupportedDatabaseException.ERROR_KEY, HttpStatus.INTERNAL_SERVER_ERROR),
           Map.entry(ERROR_KEY_UNKNOWN_ERROR, HttpStatus.INTERNAL_SERVER_ERROR));
 
-  @ExceptionHandler(MaxUploadSizeExceededException.class)
```
Comment (ryzheboka) [inline rest/taskana-rest-spring/src/main/java/pro/taskana/common/rest/TaskanaRestExceptionHandler.java:120]: Why did you change this? I couldn't find any info online about the change of @ExceptionHandler(MaxUploadSizeExceededException.class) connected to the spring update

Comment (arolfes) [reply rest/taskana-rest-spring/src/main/java/pro/taskana/common/rest/TaskanaRestExceptionHandler.java:120]: Hallo @ryzheboka ,

lässt man die Methode wie sie ist, kommt es zu dieser Exception beim Starten der Applikation:

```
aused by: java.lang.IllegalStateException: Ambiguous @ExceptionHandler method mapped for [class org.springframework.web.multipart.MaxUploadSizeExceededException]: {protected org.springframework.http.ResponseEntity pro.taskana.common.rest.TaskanaRestExceptionHandler.handleMaxUploadSizeExceededException(org.springframework.web.multipart.MaxUploadSizeExceededException,org.springframework.web.context.request.WebRequest), public final org.springframework.http.ResponseEntity org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler.handleException(java.lang.Exception,org.springframework.web.context.request.WebRequest) throws java.lang.Exception}
	at org.springframework.web.method.annotation.ExceptionHandlerMethodResolver.addExceptionMapping(ExceptionHandlerMethodResolver.java:114) ~[spring-web-6.1.5.jar:6.1.5]
	at org.springframework.web.method.annotation.ExceptionHandlerMethodResolver.<init>(ExceptionHandlerMethodResolver.java:78) ~[spring-web-6.1.5.jar:6.1.5]
```

Entfernt man die ursprüngliche Methode, funktionieren die Tests nicht mehr, da der Default von Spring greift. Deswegen musste ich die Methode hier überschreiben

**Review by ryzheboka — COMMENTED** [review]
Could you add the link to Sonarcloud?
